src/storage/StorageModule.ts

```diff
@@ -1,32 +1,111 @@
-import { BigNumberish, ethers } from "ethers";
-import { IStorageModule } from "./IStorageModule";
+import { BigNumberish } from "ethers";
+import { IStorageModule } from "./interfaces/IStorageModule";
+import {
+    StateSnapshotStruct,
+    BlockConfirmationStruct,
+    ExitChannelBlockStruct,
+    JoinChannelBlockStruct,
+    SignedBlockStruct
+} from "@typechain-types/contracts/V1/DataTypes";
+import { BlockStorageModule } from "./BlockStorageModule";
+import { IBlockStorageModule } from "./interfaces/IBlockStorageModule";
+import { IExitChannelStorageModule } from "./interfaces/IExitChannelStorageModule";
+import { IJoinChannelStorageModule } from "./interfaces/IJoinChannelStorageModule";
+import { JoinChannelStorageModule } from "./JoinChannelStorageModule";
+import { IStateSnapshotStorageModule } from "./interfaces/IStateSnapshotStorageModule";
+import { ExitChannelStorageModule } from "./ExitChannelStorageModule";
+import { StateSnapshotStorageModule } from "./StateSnapshotStorageModule";
+import { BlockHash } from "@/types/storage";
 
 export class StorageModule implements IStorageModule {
```

Comment:
I understand your logic to 'force' StorageManager to implement iStorage witch internally implements all storage interfaces, but you don't need to do that since this is the only class that get's exposed to the rest of the system so you choose the interface that is exposed and it internally knows how to interact with specific storage components and should it expose some of their APIs and to which degree 

Later if we plan on having different implementations for StorageManger it's worth creating an interface or abstract class so the concrete implementations can be polymorphically substituted - while there is only 1 instance it's just over engineering